linalg/kernels/vector.cpp

```diff
@@ -0,0 +1,260 @@
+// Copyright (c) 2010, Lawrence Livermore National Security, LLC. Produced at
+// the Lawrence Livermore National Laboratory. LLNL-CODE-443211. All Rights
+// reserved. See file COPYRIGHT for details.
+//
+// This file is part of the MFEM library. For more information and source code
+// availability see http://mfem.org.
+//
+// MFEM is free software; you can redistribute it and/or modify it under the
+// terms of the GNU Lesser General Public License (as published by the Free
+// Software Foundation) version 2.1 dated February 1999.
+
+#include "../../general/okina.hpp"
+#include "vector.hpp"
+
+// *****************************************************************************
+void kdot(const size_t N, const double *d_x, const double *d_y, double *d_dot)
```

Comment:
Why is this one outside the namespace?

Reply:
I was using cub before having removed it, it's called from `kVectorDot`, can move it back to the MFEM namespace, sure.